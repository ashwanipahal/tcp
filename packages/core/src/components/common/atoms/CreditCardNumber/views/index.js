export { default } from './CreditCardNumber.view';
