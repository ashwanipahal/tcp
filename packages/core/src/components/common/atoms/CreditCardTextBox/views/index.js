export { default } from './CreditCardTextBox.native';
