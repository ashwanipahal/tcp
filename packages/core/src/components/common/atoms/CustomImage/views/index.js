export { default } from './CustomImage.view.native';
