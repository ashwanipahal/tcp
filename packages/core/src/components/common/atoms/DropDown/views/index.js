export { default } from './DropDown.native';
