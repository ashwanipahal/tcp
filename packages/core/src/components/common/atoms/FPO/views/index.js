export { default } from './FPO.view';
