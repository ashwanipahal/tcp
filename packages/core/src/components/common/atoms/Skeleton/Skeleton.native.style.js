import styled from 'styled-components';

const getRowStyle = props => {
  const { rowProps } = props;
  return rowProps;
};

export const Row = styled.View`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  ${getRowStyle}
`;
export const Col = styled.View`
  margin-bottom: 5px;
  margin-right: 5px;
  background: #d8d8d8;
  position: relative;
  ${props => (props.outFitSkeleton ? `margin:12px 12px; ` : '')}
`;
