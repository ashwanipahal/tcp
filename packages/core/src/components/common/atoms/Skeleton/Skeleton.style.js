import { css } from 'styled-components';

const style = css`
  &.skeleton-row {
    position: relative;
  }
  .skeleton-col {
    height: 150px;
    background: #d8d8d8;
    ${props =>
      props.removeLastMargin
        ? `&:nth-child(${props.col}){
      margin-right:0;
    }`
        : ''}
  }
  .left-carousel {
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    left: 0;
    display: none;
    @media ${props => props.theme.mediaQuery.large} {
      display: block;
    }
  }
  .right-carousel {
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    right: 0;
    display: none;
    @media ${props => props.theme.mediaQuery.large} {
      display: block;
    }
  }
`;
export default style;
