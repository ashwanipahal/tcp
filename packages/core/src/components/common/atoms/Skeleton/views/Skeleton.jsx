import React from 'react';
import { PropTypes } from 'prop-types';
import withStyles from '../../../hoc/withStyles';
import style from '../Skeleton.style';
import { getIconPath } from '../../../../../utils';
import Image from '../../Image';
import { Row, Col } from '../..';

const Skeleton = ({ className, col, showArrows, colSize, removeLastMargin }) => {
  const rightCarouselPath = getIconPath('carousel-big-carrot');
  const leftCarouselPath = getIconPath('carousel-big-carrot-left');
  return (
    <Row className={`${className} skeleton-row`}>
      {Array.from({ length: col }).map((cItem, index) => {
        return (
          <Col
            key={index.toString()}
            colSize={colSize}
            className="skeleton-col"
            removeLastMargin={removeLastMargin}
          />
        );
      })}
      {showArrows && <Image src={leftCarouselPath} className="left-carousel" />}
      {showArrows && <Image src={rightCarouselPath} className="right-carousel" />}
    </Row>
  );
};

Skeleton.propTypes = {
  className: PropTypes.string.isRequired,
  rowProps: PropTypes.shape({}),
  colProps: PropTypes.shape({}),
  col: PropTypes.number,
  colSize: PropTypes.shape({}),
  showArrows: PropTypes.bool,
  removeLastMargin: PropTypes.bool,
};

Skeleton.defaultProps = {
  rowProps: {},
  colProps: {},
  col: 1,
  colSize: { large: 12, medium: 8, small: 6 },
  showArrows: false,
  removeLastMargin: false,
};

export default withStyles(Skeleton, style);
export { Skeleton as SkeletonVanilla };
