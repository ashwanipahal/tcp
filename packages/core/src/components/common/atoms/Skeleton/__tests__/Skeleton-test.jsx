import React from 'react';
import { shallow } from 'enzyme';
import { SkeletonVanilla as Skeleton } from '../views/Skeleton';

describe('Home Skeleton Componenet', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Skeleton col={3} className="skeleton" />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
