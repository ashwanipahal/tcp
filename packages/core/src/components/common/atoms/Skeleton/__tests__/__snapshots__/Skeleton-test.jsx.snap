// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Skeleton Componenet should match snapshot 1`] = `
<Styled(Row)
  className="skeleton skeleton-row"
>
  <Styled(Col)
    className="skeleton-col"
    colSize={
      Object {
        "large": 12,
        "medium": 8,
        "small": 6,
      }
    }
    key="0"
    removeLastMargin={false}
  />
  <Styled(Col)
    className="skeleton-col"
    colSize={
      Object {
        "large": 12,
        "medium": 8,
        "small": 6,
      }
    }
    key="1"
    removeLastMargin={false}
  />
  <Styled(Col)
    className="skeleton-col"
    colSize={
      Object {
        "large": 12,
        "medium": 8,
        "small": 6,
      }
    }
    key="2"
    removeLastMargin={false}
  />
</Styled(Row)>
`;
